v1.10 backports 2021-07-08

pchaigno opened the pull request:
* #16363 -- conformance tests: Use hubble-relay-ci image (@michi-covalent)
 * #16715 -- ci/conformance: Various image-related fixes (@gandro)
     - AWS-CNI workflow doesn't exist on v1.10, had to remove it.
 * #16769 -- Revert "policy: Make selectorcache callbacks lock-free" (@aanm)
 * #16767 -- test: Redeploy DNS after endpointRoutes reconfiguration (@joestringer)
 * #16804 -- contrib: Explicitly set remote for backport branches (@twpayne)
 * #16782 -- docs: account for bandwidth manager now being disabled by default (@bmcustodio)
 * #15999 -- Improve logging when cgroupfs mount fails (@johngv2)
 * #16822 -- docs: Document dns visibility limitations (@joestringer)

New PRs added:

 * #16886 -- test: do not useDigest in upstream tests (@aanm)
 * #16411 -- test: Wait for kube-dns before starting test (@jrajahalme)
 * #16735 -- workflows: remove label filters for testing workflows (@nbusseneau)
 * #16711 -- workflows: fix concurrency group names (@nbusseneau)
 * #16787 -- workflows: various fixes & consistency passes (@nbusseneau)
 * #16830 -- workflows: fix L4LB test missing PR reporting on issue_comment (@nbusseneau)
 * #16831 -- workflows: fix Relay pgrep check when using additional flags (@nbusseneau)
 * #16799 -- Bump cilium-cli to v0.8.4 (@tklauser)
 * #16834 -- envoy: Keep track of proxy listeners separately (@jrajahalme)
 * #16440 -- test: Delete Istio resources if install does not complete (@jrajahalme)
 * #16819 -- github: Increase workflow timeout (@jrajahalme)
 * #16755 -- docs(k3s): add back the flag to disable network policies (@rio)
 * #16845 -- test: Move instrumentation to AfterFailed instead of AfterAll (@christarazi)
 * #16835 -- test: Delete DNS pods in AfterAll for datapath tests (@joestringer)
 * #16817 -- iptables: Remove leading zeroes (@jrajahalme)
 * #16801 -- Fix potential deadlock in pod identity updates (@jrajahalme)
 * #16857 -- pkg/k8s: re-add CiliumIsUp Node condition even if removed (@aanm)

~This pull request also reverts https://github.com/cilium/cilium/pull/16824 because it breaks the k8s-upstream CI job.~

Skipped because the PR it fixes (which introduces the new test file) is not backported yet:

 * #16775 -- test/Bookinfo: Collect full artifact in case of failure (@pchaigno)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 16363 16715 16769 16767 16775 16804 16782 15999 16822 16886 16411 16735 16711 16787 16830 16831 16799 16834 16440 16819 16755 16845 16835 16817 16801 16857; do contrib/backporting/set-labels.py $pr done 1.10; done
```

michi-covalent committed:
conformance tests: Use hubble-relay-ci image

[ upstream commit 915379ede7cd98261ee2143d8986264085d92cc6 ]

Signed-off-by: Michi Mutsuzaki <michi@isovalent.com>
Signed-off-by: Paul Chaignon <paul@cilium.io>

jrajahalme committed:
test: Wait for kube-dns before starting test

[ upstream commit d790f8f5f06c5e9fe90bd0ba9b98b5488dbae469 ]

Wait for kube-dns to become reachable before running test in fqdn.go.

Fixes: #16409
Signed-off-by: Jarno Rajahalme <jarno@isovalent.com>
Signed-off-by: André Martins <andre@cilium.io>

jrajahalme committed:
policy: Make selectorcache callbacks lock-free

[ upstream commit 7e91f36c5c9845af8de62a652a5406c206b0bb24 ]

Make IdentitySelectionUpdated() callbacks lock-free by queueing them
while still holding selectorcache lock (to keep FIFO order) and
calling from a goroutine not holding any locks. This prevents
deadlocks caused by the implementation of IdentitySelectionUpdated()
taking locks such as endpoint or selectorcache locks.

Signed-off-by: Jarno Rajahalme <jarno@isovalent.com>
Signed-off-by: André Martins <andre@cilium.io>

jrajahalme committed:
test: Delete Istio resources if install does not complete

[ upstream commit 99c44dc51c55e01986f63aa68a0bb9879efca61a ]

If Istio install fails (e.g., times out) during BeforeAll(), the
resource cleanups in AfterAll() are not performed, so have to clean up
explicitly after failure in BeforeAll instead.

Without this some Istio resources may be left behind, causing other
tests to fail due to unexpected pods being present.

Signed-off-by: Jarno Rajahalme <jarno@isovalent.com>
Signed-off-by: André Martins <andre@cilium.io>

nbusseneau committed:
workflows: remove Azure SP from cilium install args

[ upstream commit f22e086541dc1fbd1618703b8b8c5a3f0019e8be ]

Following modifications to auto-generated Azure Service Principal in the
CLI, we can stop passing the main Azure SP to `cilium install`:
https://github.com/cilium/cilium-cli/pull/308

The auto-generated Service Principal is more secure as it is unique and
only has access to the cluster used for a specific workflow run.

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>
Signed-off-by: André Martins <andre@cilium.io>

nbusseneau committed:
workflows: replace port-forwards with dedicated hubble command

[ upstream commit bae1a65cf68d19115cd88deb181a3acb7f1ff775 ]

Also refactor Cilium cleanup for a cleaner `pkill` pass.

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>
Signed-off-by: André Martins <andre@cilium.io>

nbusseneau committed:
workflows: remove unnecessary relay deployment in 2nd cluster

[ upstream commit 45d71524a6afd83465f09854f8a739388c0887c1 ]

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>
Signed-off-by: André Martins <andre@cilium.io>

nbusseneau committed:
workflows: standardize/comment all post-test and cleanup operations

[ upstream commit 2c77199adf368008d9e9a350b45e75a17344a1bc ]

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>
Signed-off-by: André Martins <andre@cilium.io>

nbusseneau committed:
workflows: remove connectivity test pods restart workaround

[ upstream commit 464f97c67ed4a3f7bf0e945c86e2f0dc9c3ed00b ]

The flag `--force-deploy` does the same :)

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>
Signed-off-by: André Martins <andre@cilium.io>

nbusseneau committed:
workflows: consistently disabling flow validation

[ upstream commit 36aa4f781f440af6678647f4428391e1a02889fb ]

In #16388, we disabled flow validation as it was unstable in the CLI.
In #16617, we partially moved to `warning` for testing purposes.

This reverts back to `disabled` for all connectivity tests in order to
be consistent. Also fix missing disabled flow validation in EKS, as it
seems like we missed that one in #16388.

We can always enable flow validation again later once we are confident
flow validation works better in the CLI.

Fixes 49ce8e9953b877e6a9b2b1c71df1b767837cb690
Fixes 21c9b41c5ab5f580fa94591de9427f3a0965be35

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>
Signed-off-by: André Martins <andre@cilium.io>

nbusseneau committed:
workflows: properly setup multicluster cross-cluster traffic

[ upstream commit c4bdab1341ae4987f332134bde1628442c894953 ]

- Both clusters created as VPC-native.
- Use common native routing CIDR.
- Cluster 2 inherits CA from cluster 1.
- Add dedicated firewall rule.

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>
Signed-off-by: André Martins <andre@cilium.io>

nbusseneau committed:
workflows: fix concurrency group names

[ upstream commit d1af0d3ad89a2570837c9f4387241f504f64c186 ]

Improve concurrency group names uniqueness, in particular for allowing
testing workflow changes via `pull_request` events.

In the previous version, all `pull_request`-triggered runs would end up
in the same concurrency group as the `scheduled` events, due to not
having a `github.event.issue` object.

This new version proposes a new structure that should be unique for all
types of testing, while still allowing runs of the same type to override
each other:

Structure:
- Workflow name
- Event type
- A unique identifier depending on event type:
  - schedule: SHA
  - issue_comment: PR number
  - pull_request: PR number

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>
Signed-off-by: André Martins <andre@cilium.io>

gandro committed:
ci-multicluster: Pull in correct image version

[ upstream commit 763b88b11291ff8830fe2df70495bbd0ce406ede ]

Pull in the correct image version built as part of CI when deploying
`clustermesh-apiserver`.

Signed-off-by: Sebastian Wicki <sebastian@isovalent.com>
Signed-off-by: Paul Chaignon <paul@cilium.io>

gandro committed:
ci/conformance: Wait for hubble-relay-ci image

[ upstream commit 8cfb1f89612bc8ffa9b6f23d9eae43276e951604 ]

This commit extends the conformance tests (all of which are using
`cilium hubble enable` to deploy Hubble Relay) to also wait for the
`hubble-relay-ci` image to be built.

In the multicluster test, we also wait for the
`clustermesh-apiserver-ci` image.

Signed-off-by: Sebastian Wicki <sebastian@isovalent.com>
Signed-off-by: Paul Chaignon <paul@cilium.io>

Commit:
cgroups: Fix improper error on return

[ upstream commit e3f9c61a67815fcbb22bc545267496d8372bd218 ]

If cgroupv2 is not mounted, a logical mistake returns a error
even after a successful mount, which is silently ignored by
the caller function.

This commit fixes the logical mistake and takes care of print a warning
message if the cgroupv2 could not be mounted.

Fixes: #15997

Signed-off-by: Joao Victorino <joao@accuknox.com>
Signed-off-by: Paul Chaignon <paul@cilium.io>

nbusseneau committed:
workflows: remove label filters for testing workflows

[ upstream commit 512131878bb733cb335761ff73384d3d0cf127eb ]

Previously, we used `pull_request` `labeled` events for testing changes
to workflows from PRs. This was actually a remnant of our first CI 3.0
implementation to be able to run workflows on-demand rather than on
every PR push, before we switched to `issue_comment` triggers.

We remove the `labeled` filter as they only make it way more complicated
for contributors to test workflow changes. Instead, we switch to a more
simple workflow where simply uncommenting the `pull_request` trigger is
enough to be able to start testing changes to a workflow.

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>
Signed-off-by: André Martins <andre@cilium.io>

rio committed:
docs(k3s): add back the flag to disable network policies

[ upstream commit 7bd301bb5606605114512e445901f4f5febeec8d ]

This was actually added in https://github.com/cilium/cilium/pull/13783 but got removed again after some other changes.
That flag is required because otherwise any liveness or readiness probes will be classified as *world* traffic and get
dropped if network policies are installed.

Signed-off-by: Rio Kierkels <riokierkels@gmail.com>
Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
Revert "policy: Make selectorcache callbacks lock-free"

[ upstream commit a97bd0d8f99fa18be533570b9f6afa4ce4649f3a ]

This reverts commit a75599da7964fb5e24c3362dfbdedf7d2f455089.
has it seems to be causing a lot of FQDN flakes through the entire CI.

Signed-off-by: André Martins <andre@cilium.io>
Signed-off-by: Paul Chaignon <paul@cilium.io>

bmcustodio committed:
docs: bandwidth manager is now disabled by default

[ upstream commit 1abf16b8622a7b3908b229479646deb7af14d36f ]

https://github.com/cilium/cilium/pull/16380 disabled the bandwidth
manager by default.

Signed-off-by: Bruno Miguel Custódio <brunomcustodio@gmail.com>
Signed-off-by: Paul Chaignon <paul@cilium.io>

bmcustodio committed:
docs: add missing words to the spelling checklist

[ upstream commit 7a22ee8a533b80cf232bfe79913b03dc331c2009 ]

Signed-off-by: Bruno Miguel Custódio <brunomcustodio@gmail.com>
Signed-off-by: Paul Chaignon <paul@cilium.io>

jrajahalme committed:
wip: Add WaitGroup for SelectorCache user notifications

[ upstream commit fc6ef4d5cd0764c7e67a72ed62b105e4c1c80263 ]

Signed-off-by: Jarno Rajahalme <jarno@isovalent.com>
Signed-off-by: André Martins <andre@cilium.io>

twpayne committed:
contrib: Explicitly set remote for backport branches

[ upstream commit eea7f9c3903da135675c0d2e1566067a89706e10 ]

Before this change, the backporting scripts would correctly push to the
user's remote, but the default remote for the backport branch was left
as origin (typically cilium/cilium).

This commit sets the backporting branch's remote to the user's remote,
so further pushes from the command line (e.g. after adding more commits
or rebasing) go by default to the correct remote.

Signed-off-by: Tom Payne <tom@isovalent.com>
Signed-off-by: Paul Chaignon <paul@cilium.io>

nbusseneau committed:
workflows: fix tests ignore for multicluster testing

[ upstream commit 45c9c4d51151a40295d94b724a2488a37c3273f2 ]

We changed that in v0.8 of the CLI: https://github.com/cilium/cilium-cli/pull/215

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>
Signed-off-by: André Martins <andre@cilium.io>

joestringer committed:
test: Redeploy DNS after changing endpointRoutes

[ upstream commit a0e77127dcd7165ed1b07e3c800e41f74a342cbe ]

In general up until now, Cilium has expected endpointRoutes mode to be
set to exactly one value upon deployment and for that value to stay the
same for the remainder of operation. Toggling it can lead to a mix of
endpoints in different datapath modes which is not well covered in CI.

In Github issue #16717 we observed that if the testsuite toggles this
setting then we can end up with kubedns pods remaining in endpoint
routes mode, even though the rest of the daemon (and other pods) are not
configured in this mode. This can lead to connectivity issues in DNS,
and a range of test failures in subsequent tests because DNS is broken.

Longer term to resolve this, we could improve on Cilium to ensure that
users can successfully toggle this setting on or off at runtime and
properly handle this case, or alternatively shift all logic over to
endpoint-routes mode by default and disable the other option.

Given that CI for the master branch is in a poor state due to this issue
today, and that part of the issue is CI reconfiguring the datapath state
of Cilium during the test setup in an unsupported manner, this commit
proposes to force DNS pod redeployment as part of setup any time a test
reconfigures the endpointRoutes mode. This should mitigate the testing
side issue while we mull over the right longer-term solution.

Signed-off-by: Joe Stringer <joe@cilium.io>
Signed-off-by: Paul Chaignon <paul@cilium.io>

jrajahalme committed:
iptables: Remove leading zeroes

[ upstream commit d5ff6879dbc50de93cde07b4e6c87f2581106f34 ]

Remove leading zeroes from marks, as 'iptables' is not formatting
them. This allows proper matching of existing rules and avoids
appending duplicate rules.

Signed-off-by: Jarno Rajahalme <jarno@isovalent.com>
Signed-off-by: André Martins <andre@cilium.io>

jrajahalme committed:
github: Increase workflow timeout

[ upstream commit 72ba43d0219d1f2fb6a37e4f86c49a1cd18b1f63 ]

Increase conformance-gke timeout to avoid fail due to cancelled test
as in:

installation-and-connectivity
cancelled 2 hours ago in 30m 13s

[=] Test [client-egress]
....
[=] Test [echo-ingress]
Error: The operation was canceled.

Signed-off-by: Jarno Rajahalme <jarno@isovalent.com>
Signed-off-by: André Martins <andre@cilium.io>

nbusseneau committed:
workflows: check status of Relay port-forward

[ upstream commit 00ae9fb492f0c6313277106bdf50c740d0d395cc ]

Since we start a background process for port-forwarding Relay, we must
check if it actually succeeded or failed before allowing workflows to
continue.

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>
Signed-off-by: André Martins <andre@cilium.io>

joestringer committed:
docs: Document dns visibility limitations

[ upstream commit 9c8004486975100c133fb0ed3b5321dd822f2efd ]

Signed-off-by: Joe Stringer <joe@cilium.io>
Signed-off-by: Paul Chaignon <paul@cilium.io>

nbusseneau committed:
workflows: fix L4LB test missing PR reporting on issue_comment

[ upstream commit d2da30ae18e8314ff7473936632e6dc964001848 ]

Unlike `pull_request`-triggered runs, `issue_comment`-triggered runs
cannot automatically report back to the PR they were triggered from.

Adding manual reporting as per the other workflows.

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>
Signed-off-by: André Martins <andre@cilium.io>

nbusseneau committed:
workflows: fix Relay pgrep check when using additional flags

[ upstream commit 41b821d573b8ac46d2bd25e5e986e77bcc203c16 ]

When passing additional flags to `cilium hubble port-forward`, both the
original command and `kubectl` child process might be conflated with
additional flags.

We add wildcards for matching the additional flags to both `pkill`
cleanups and `pgrep` port-forward success checks.

Fixes 00ae9fb492f0c6313277106bdf50c740d0d395cc
Follow-up to #16787

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>
Signed-off-by: André Martins <andre@cilium.io>

jrajahalme committed:
envoy: Keep track of proxy listeners separately

[ upstream commit 099c34d977b73491618454d1a9ea253623665c2d ]

Since the addition of Envoy prometheus listener it has been possible
to have non-proxy listeners configured with Envoy. Waiting for Envoy
N/ACKs must be disabled when no proxy listeners are configured, even
if a prometheus listener may still be configured.

Without this fix adding endpoints may fail due to not receiving N/ACKs
from Envoy after Envoy has been started due to an L7 network policy,
and this policy is removed, if the Cilium option
'--proxy-prometheus-port' is also configured.

Fixes: #12949
Signed-off-by: Jarno Rajahalme <jarno@isovalent.com>
Signed-off-by: André Martins <andre@cilium.io>

joestringer committed:
test: Redeploy DNS pods in AfterAll for datapath tests

[ upstream commit c18cfc874620163769f0f66863b1c15962d25e1f ]

Commit a0e77127dcd7 ("test: Redeploy DNS after changing endpointRoutes")
didn't go quite far enough: It ensured that between individual tests in
a given file, the DNS pods would be redeployed during the next run if
there were significant enough datapath changes. However, the way it did
this was by storing state within the 'kubectl' variable, which is
recreated in each test file. So if the last test in one CI run enabled
endpoint routes mode, then the DNS pods would not be redeployed to
disable endpoint routes mode as part of the next test.

Fix it by redeploying DNS after removing Cilium from the cluster.
Kubernetes will remove the current DNS pods and reschedule them, but
they will not launch until the next test deploys a new version of
Cilium.

Reported-by: Chris Tarazi <chris@isovalent.com>
Fixes: 0e77127dcd7 ("test: Redeploy DNS after changing endpointRoutes")
Related: #16717

Signed-off-by: Joe Stringer <joe@cilium.io>
Signed-off-by: André Martins <andre@cilium.io>

joestringer reviewed (approved):
:+1: There'll be a followup for #16767 but that can be handled in a subsequent backport or tacked on top.

aditighag reviewed (approved):
>Improve logging when cgroupfs mount fails #15999 -- Improve logging when cgroupfs mount fails (@johngv2)

Looks good! Thanks!

tklauser committed:
.github/workflows: bump cilium-cli to v0.8.4

[ upstream commit 4e3e5b831ffce013b6f58f66fcfee01fe3e34c8c ]

Signed-off-by: Tobias Klauser <tobias@cilium.io>
Signed-off-by: André Martins <andre@cilium.io>

christarazi committed:
test: Move instrumentation to AfterFailed instead of AfterAll

[ upstream commit 9744ff0131d5a8e5d7fb4643a782fc11207deb0a ]

When inspecting a sysdump of this failing test, the expected artifacts
containing the instrumentation (debug-events) are not present. This is
because AfterAll is not called when the test fails. The logic should be
in AfterFailed.

Fixes: e63aa2b ("test: Instrument LB IP via BGP test with debug-events")
Fixes: https://github.com/cilium/cilium/pull/16445

Signed-off-by: Chris Tarazi <chris@isovalent.com>
Signed-off-by: André Martins <andre@cilium.io>

gandro reviewed (approved):
Looks good for the CI 3.0 related commits. Thanks!

bmcustodio reviewed (approved):
My commit LGTM!

aanm committed:
pkg/k8s: re-add CiliumIsUp Node condition even if removed

[ upstream commit 54ae42b24669d81b18c65ca9a19fe9934dbfbbea ]

If the CiliumIsUp Node condition is removed either by accident or due
a concurrency issue when updating the node from different entities,
Cilium is now able to re-add the node condition back to the node again.

Fixes: bd34b95a7939 ("pkg/k8s: remove node.cilium.io/agent-not-ready taint from nodes")
Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
test: do not useDigest in upstream tests

[ upstream commit 3c64049af04134fcdd31b42ece4b46ac0397975e ]

Upstream tests should be executed against the code being submitted
in a PR. If `useDigest` is set to true it will use the default digest
defined in the helm charts, which in stable branches is the last
stable release and not the code being submitted in a PR.

Fixes: 41f66a1eb4fa (".github: use quay.io images in smoke tests")
Signed-off-by: André Martins <andre@cilium.io>

qmonnet reviewed (approved):
Reviewing for:
> #16755 -- docs(k3s): add back the flag to disable network policies (`@rio`)

The commit looks good to me

jrajahalme reviewed (approved):
Backports for all of my commits seem to be done.

nbusseneau reviewed (changes requested):
- Commit from #16711 seems to be missing.
- Commit from #16830 seems to be missing.
- One commit of #16787 (f98a99de3ae4afb9f22065d08a5cf5a7be168449) seems to be missing.

Other commits LGTM.

nbusseneau reviewed (approved):
🚀
